### PR TITLE
[DOCS] DataContext and CheckpointConfig DocString

### PR DESCRIPTION
**Pull request by superengi**
Added the following DocStrings

- DataContext
- S3StoreBackendDefaults *
- FilesystemStoreBackendDefaults *
- CheckpointConfig
- ValidationResultIdentifier *
- InMemoryStoreBackendDefaults *

Items marked with a (*) had an existing docstring, but I've added the public_api declaration onto it.

### CODE REVIEW

File: great_expectations/data_context/data_context/data_context.py [hunk 1]
```diff
@@ -66,7 +63,7 @@ def DataContext(
 
 # TODO: add additional overloads
 
-
+@public_api
 def DataContext(
```
Comment (anthonyburdi) [inline great_expectations/data_context/data_context/data_context.py:73]: How about adding the `@deprecated_argument` decorator for the `ge_cloud_*` arguments? Here is an example usage of this decorator on `add_checkpoint()`: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/data_context/data_context/abstract_data_context.py#L1295

File: great_expectations/data_context/data_context/data_context.py [hunk 2]
```diff
@@ -80,22 +77,20 @@ def DataContext(
...
-    Data Context that may be better suited for your particular deployment e.g. ephemerally or backed by GX Cloud
-    (coming soon). Please refer to our documentation for more details.
+    It is the primary entry point for a Great Expectations deployment, with configurations and methods for all
+    supporting components. The DataContext is configured via a yml file stored in a directory called great_expectations;
+    this configuration file as well as managed Expectation Suites should be stored in version control. There are other
+    ways to create a Data Context that may be better suited for your particular deployment e.g. ephemerally or backed by
+    GX Cloud (coming soon). Please refer to our documentation for more details.
 
     You can Validate data or generate Expectations using Execution Engines including:
-
```
Comment (anthonyburdi) [inline great_expectations/data_context/data_context/data_context.py:92]: I think these line breaks may be necessary before a bulleted list, but take a look at the rendering to confirm.

File: great_expectations/data_context/data_context/data_context.py [hunk 3]
```diff
@@ -112,6 +107,19 @@ def DataContext(
     --Documentation--
         - https://docs.greatexpectations.io/docs/terms/data_context
 
+    Args:
+        context_root_dir: The context root directory
+        runtime_environment: The runtime environment
+        cloud_mode: The cloud mode
```
Comment (anthonyburdi) [inline great_expectations/data_context/data_context/data_context.py]: How about adding more description to these args about what they are?

File: great_expectations/data_context/data_context/data_context.py [hunk 4]
```diff
@@ -112,6 +107,19 @@ def DataContext(
...
+        cloud_base_url: Your cloud base url
+        cloud_access_token: Your cloud access token
+        cloud_organization_id: Your cloud organization ID
+        ge_cloud_mode: The deprecated cloud mode
+        ge_cloud_base_url: Your deprecated cloud base url
+        ge_cloud_access_token: Your deprecated cloud access token
+        ge_cloud_organization_id: Your deprecated cloud organization ID
+    Returns:
+        context
```
Comment (anthonyburdi) [inline great_expectations/data_context/data_context/data_context.py]: There needs to be a line break here between the last arg and the `Returns:` line.
```suggestion
        ge_cloud_organization_id: Your deprecated cloud organization ID

    Returns:
        context
```

**Review by anthonyburdi — COMMENTED** [review]
Thank you for writing these docstrings! I made a few comments while figuring out what the linter errors were, will need to do a more thorough review later but submitting them for now in case they are helpful.

**Review by anthonyburdi — APPROVED** [review]

**Comment by superengi** [comment]
@cla-bot check

**Pull request merged**